tests/test_rest_api.py

```diff
@@ -273,12 +280,18 @@ def test_ds_dict_cache(ds_dict):
 def test_single_dataset_openapi_override(airtemp_rest):
     openapi_schema = airtemp_rest.app.openapi()
 
-    # "dataset_id" parameter should be absent in all paths
-    assert len(openapi_schema['paths']['/']['get']['parameters']) == 0
-
-    # test cached value
-    openapi_schema = airtemp_rest.app.openapi()
-    assert len(openapi_schema['paths']['/']['get']['parameters']) == 0
+    with pytest.raises(KeyError):
+        # "dataset_id" parameter should be absent in all paths
+        # parameters is no longer generated when plugins use passed in deps
+        #  and get_dataset is replaced
+        assert len(openapi_schema['paths']['/']['get']['parameters']) == 0
+
+    with pytest.raises(KeyError):
+        # test cached value
+        # parameters is no longer generated when plugins use passed in deps
+        #  and get_dataset is replaced
+        openapi_schema = airtemp_rest.app.openapi()
+        assert len(openapi_schema['paths']['/']['get']['parameters']) == 0
```

Comment:
With the change in dependency loading for built in routes (via plugins), `get_dataset` doesn't take an argument for single datasets, and thus the `parameters` key doesn't get added to the openapi spec. 

It still would get added (and need to be overridden) for any non-plugin routers, so the modified spec generator is still needed for single datasets.